src/components/popover/popover-trigger/popover-trigger.ts

```diff
@@ -47,22 +33,17 @@ export class SbbPopoverTriggerElement extends LitElement {
     }
   }
 
-  public override disconnectedCallback(): void {
-    super.disconnectedCallback();
-    this._handlerRepository.disconnect();
-  }
-
-  protected override render(): TemplateResult {
-    const { hostAttributes } = resolveButtonRenderVariables(this);
-
-    setAttributes(this, hostAttributes);
-
+  protected override renderIconSlot(): TemplateResult {
     return html`
-      <span class="sbb-popover-trigger">
-        <slot>${this.iconName ? html`<sbb-icon name=${this.iconName}></sbb-icon>` : nothing}</slot>
-      </span>
+      <slot>
```

Comment:
Nitpick; I think this code could just be directly moved to the renderTemplate method.